feature-libs/asm/components/customer-selection/customer-selection.component.html

```diff
@@ -7,33 +7,49 @@
       formControlName="searchTerm"
       [attr.aria-label]="'asm.customerSearch.searchTerm.label' | cxTranslate"
       placeholder="{{ 'asm.customerSearch.searchTerm.label' | cxTranslate }}"
+      (keydown.arrowdown)="focusFirstItem($event)"
+      (keydown.end)="setSelectionEnd($event)"
     />
     <cx-form-errors
       [control]="customerSelectionForm.get('searchTerm')"
     ></cx-form-errors>
   </label>
-  <button *cxFeatureLevel="'!5.1'" type="submit">
-    {{ 'asm.customerSearch.submit' | cxTranslate }}
-  </button>
-  <button
-    *cxFeatureLevel="'5.1'"
-    type="submit"
-    [class.active]="selectedCustomer"
-  >
+  <button type="submit" [class.active]="selectedCustomer">
     {{ 'asm.customerSearch.submit' | cxTranslate }}
   </button>
 </form>
 
 <div *ngIf="searchResults | async as results" class="asm-results" #resultList>
   <button
-    *ngFor="let result of results.entries"
-    (click)="selectCustomerFromList(result)"
+    #searchResultItem
+    *ngFor="let result of results.entries; let i = index"
+    [tabindex]="activeFocusedButtonIndex === i ? 0 : -1"
+    [attr.aria-selected]="activeFocusedButtonIndex === i"
+    [class.active]="activeFocusedButtonIndex === i"
+    (keydown.arrowup)="focusPreviousChild($event)"
+    (keydown.arrowdown)="focusNextChild($event)"
+    (keydown.arrowright)="focusInputText($event)"
+    (keydown.arrowleft)="focusInputText($event)"
+    (keydown.enter)="selectCustomerFromList($event, result)"
+    (keydown.escape)="closeResults($event)"
+    (keydown.home)="focusInputText($event)"
+    (keydown.end)="focusInputText($event)"
```

Comment:
Feels like it would be cleaner to put these listeners on the parent element (`div.asm-results`).

Reply:
this button element is in *ngFor and button itself has focus and click so we need to handle separately if we move to parent so  i think it is better to stay here

Reply:
Fair point on the button itself having focus, but the click event would propagate to the parent element. It's not uncommon to write Javascript this way.

Reply:
also, no search result button needs to be moved out from the parent so i will keep what i have